diff --git a/README b/README
index 7f0ec56..9c06c84 100644
--- a/README
+++ b/README
@@ -17,8 +17,79 @@
 EPICS Verions:
 --------------
 This driver has currently been built and tested at SLAC with EPICS 3.14.8.2.
-This driver has implimented EPICS OSI function calls and is OSI complient.
+This driver has implemented EPICS OSI function calls and is OSI compliant.
 
 Operating System:
 -------------------
 This driver has only been tested  under RTEMS. 
+
+How to use this module in your IOC
+------------------
+1.	Add the following lines in your iocTop/configure/RELEASE:
+IP231=$(EPICS_MODULES)/ip231/$(IP231_MODULE_VERSION)
+
+2.	Copy the following files to your iocTop/<iocName>/<iocName>App/Db directory:
+	ip231App/Db/IP231.substitutions
+
+3.	Add the following lines in your iocTop/<iocName>/<iocName>App/Db/Makefile:
+DB += IP231.db
+
+4.	Add the following lines in your iocTop/<iocName>/<iocName>App/src/Makefile:
+<iocName>_DBD += IP231.dbd
+<iocName>_LIBS += IP231
+
+5.	Add the following lines in your iocTop/<iocName>/<iocName>App/src/<iocName>Main.cpp:
+//  Include these lines to force linker to include xy9660 Ipac carrier
+//	Note: There are other ipac carrier modules besides the xy9660.
+//	See the ipac module for options.
+#include "drvXy9660.h"
+extern "C"
+int addCarrier( const char * cardParams )
+{
+    return ipacAddXVME9660( cardParams );
+}
+
+6.	Add the following lines in your iocTop/<iocName>/iocBoot/ioc<iocName>/st.cmd:
+# Initialize IPAC carrier
+# The 2nd parameter is the base I/O address set by the jumpers on your carrier board
+# Make sure this address doesn't conflict with other boards.
+addCarrier( "0x0000" )
+
+# Initialize IP231 analog output module
+# ip231Create( <name>, <card>, <slot>, <dacmode> )
+# <card> is 0 for the first ipac carrier
+# <slot> is 0..3 for carrier slots A..D
+# <dacmode> must be "transparent" or "simultaneous"
+# In transparent mode, outputs are updated when written
+# In simultaneous mode, multiple output values can be setup,
+# and all outputs update when ip231SimulTrigger(<cardNo>) is called
+ip231Create( "ao0", 0, 3, "transparent" )
+
+# Load the EPICS records
+dbLoadRecords("db/IP231.db","PRE=TST:AOUT:,CARD=ao0")
+
+############################################################
+
+EPICS PV Values:
+----------------------
+The default names assigned by the db template and substitution files
+in this module are shown below.  These can be customized after you copy the
+template and substitution files to your ioc.
+------------------
+PV Name:	Type:		Description:
+TST:AOUT:0	ao			Analog Output 0
+TST:AOUT:1	ao			Analog Output 1
+TST:AOUT:2	ao			Analog Output 2
+TST:AOUT:3	ao			Analog Output 3
+TST:AOUT:4	ao			Analog Output 4
+TST:AOUT:5	ao			Analog Output 5
+TST:AOUT:6	ao			Analog Output 6
+TST:AOUT:7	ao			Analog Output 7
+TST:AOUT:8	ao			Analog Output 8
+TST:AOUT:9	ao			Analog Output 9
+TST:AOUT:10	ao			Analog Output 10
+TST:AOUT:11	ao			Analog Output 11
+TST:AOUT:12	ao			Analog Output 12
+TST:AOUT:13	ao			Analog Output 13
+TST:AOUT:14	ao			Analog Output 14
+TST:AOUT:15	ao			Analog Output 15
